linode/manager
# feat: [M3-8054] - Create Linode - Create Using Command Line (DX Tools Additions)

## Description 📝
This PR introduces Integrations tab and SDK's tab to ApiAwarenessModal, it generates code snippets for Terraform, Ansible, Go and Python.  

Note: Interfaces are not added to go snippets as it is getting more complex in generating the code snippet, this involves in defining the go variable and using within generated snippets. This will be tackled in a separate ticket. 

## Changes  🔄
List any change relevant to the reviewer.
- Add new tabs to ApiAwarenessModal
- New utils to generate code snippets
- Refactored ApiAwarenessModal. 

## Target release date 🗓️
7/08

## Preview 📷
| Integrations - Ansible  | Integrations - Terraform  |
| ------- | ------- |
|![image](https://github.com/linode/manager/assets/119517080/42cca325-86ba-423b-b294-3d6de4b604d0)| ![image](https://github.com/linode/manager/assets/119517080/12dcd26b-e85b-4c94-88f8-a8f4b668fde2)|

| SDK - GO  | SDK - Python  |
| ------- | ------- |
|![image](https://github.com/linode/manager/assets/119517080/030ff0fc-2dcf-442e-b37d-802f27b148cb)|![image](https://github.com/linode/manager/assets/119517080/660e03ad-8e31-4556-b9a2-db185f6d0dcf)|

## How to test 🧪
### Verification steps
(How to verify changes)
- Navigate to Linode create flow, click on "View Code Snippets" and turn on feature flag. 
- Verify the new tabs in ApiAwarenessModal.
- Verify the UX, copy changes, and links.
- Verify the copy and pate functionality of each code snippet generated.

## As an Author I have considered 🤔

*Check all that apply*
